src/components/DateRangePicker.js

```diff
@@ -152,12 +156,49 @@ class DateRangePicker extends React.Component {
 
   _handleDefaultRangeSelection = (range) => {
     this.props.onDateSelect(range.getStartDate(), range.getEndDate(), range.displayValue);
+    this.setState({ focusedDay: range.getEndDate() });
 
     if (this.props.closeCalendarOnRangeSelect) {
       this._handleScrimClick();
     }
   };
 
+  _handleDayKeyDown = (e) => {
+    const startDate = moment.unix(this.state.currentDate).startOf('month').startOf('week');
+    const endDate = moment.unix(this.state.currentDate).endOf('month').endOf('week');
+
+    if (keycode(e) === 'right') {
+      if (moment.unix(this.state.focusedDay).add(1, 'days').startOf('day').isSameOrAfter(endDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).add(1, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).add(1, 'days').startOf('day').unix() });
+    } else if (keycode(e) === 'left') {
+      if (moment.unix(this.state.focusedDay).subtract(1, 'days').startOf('day').isBefore(startDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).subtract(1, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).subtract(1, 'days').startOf('day').unix() });
+    } else if (keycode(e) === 'enter') {
+      this._handleDateSelect(true, this.state.focusedDay);
+    } else if (keycode(e) === 'up') {
+      e.preventDefault(); //stop browserscrolling
+      if (moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day').isBefore(startDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day').unix() });
+    } else if (keycode(e) === 'down') {
+      e.preventDefault(); //stop browserscrolling
+
+      if (moment.unix(this.state.focusedDay).add(7, 'days').startOf('day').isSameOrAfter(endDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).add(7, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).add(7, 'days').startOf('day').unix() });
```

Comment:
Is it necessary to call `.unix()` at the end?

Reply:
yes because i want to pass around unix time stamps instead of moment objects - i think there's less room for error/mutation this way but lmk if you have other thoughts

Reply:
That makes sense 👍